diracx-logic/tests/logic/test_sandboxes.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+import hashlib
+import secrets
+from datetime import timedelta
+from io import BytesIO
+from typing import AsyncGenerator, Generator
+
+import botocore.exceptions
+import freezegun
+import httpx
+import pytest
+import sqlalchemy
+
+from diracx.core.exceptions import SandboxNotFoundError
+from diracx.core.models import ChecksumAlgorithm, SandboxFormat, SandboxInfo, UserInfo
+from diracx.core.settings import SandboxStoreSettings
+from diracx.db.sql.sandbox_metadata.db import SandboxMetadataDB
+from diracx.logic.jobs.sandboxes import (
+    clean_sandboxes,
+    get_sandbox_file,
+    initiate_sandbox_upload,
+)
+from diracx.testing.time import mock_sqlite_time
+
+FAKE_USER_INFO = UserInfo(
+    sub="fakevo:97ae90d3-36aa-4271-becf-e61173d93fe3",
+    preferred_username="fakeuser",
+    dirac_group="fake_group",
+    vo="fakevo",
+)
+
+
+@pytest.fixture
+async def sandbox_metadata_db() -> AsyncGenerator[SandboxMetadataDB, None]:
+    """Create a fake sandbox metadata database."""
+    db = SandboxMetadataDB(db_url="sqlite+aiosqlite:///:memory:")
+    async with db.engine_context():
+        sqlalchemy.event.listen(db.engine.sync_engine, "connect", mock_sqlite_time)
+
+        async with db.engine.begin() as conn:
+            await conn.run_sync(db.metadata.create_all)
+
+        yield db
+
+
+@pytest.fixture
+async def sandbox_settings(
+    test_sandbox_settings,
+) -> AsyncGenerator[SandboxStoreSettings, None]:
+    """Create a fake sandbox settings."""
+    async with test_sandbox_settings.lifetime_function():
+        yield test_sandbox_settings
+
+
+@pytest.fixture()
+def frozen_time() -> Generator[freezegun.FreezeGun, None]:
```

Comment:
I wonder if this fixture kicks in before the databases are created and e.g. other fixtures such as the one which insert valid jobs into the test DBs?

Reply:
There is no guarantees around the order by default:

https://docs.pytest.org/en/stable/reference/fixtures.html#fixture-instantiation-order

The solution would be to make the `valid_job_ids` depend on the frozen time or use a function instead of a fixture and just call it inside the test function.